src/GDGUkraine/lib/tools/authorize.py

```diff
@@ -0,0 +1,38 @@
+import cherrypy
+
+
+__all__ = ['AuthorizeTool']
+
+
+class AuthorizeTool(cherrypy.Tool):
+
+    def __init__(self):
+        cherrypy.Tool.__init__(
+            self, 'before_handler', self._fetch, priority=20
+        )
+
+    def _setup(self):
+        cherrypy.Tool._setup(self)
+        cherrypy.request.hooks.attach(
+            'on_end_resource', self._cleanup, priority=80
+        )
+
+    def _fetch(self):
+        session = cherrypy.session
+
+        google_oauth_token = session.get('google_oauth_token')
+        google_user = session.get('google_user')
+        admin_user = session.get('admin_user')
+
+        if not google_user:
+            raise cherrypy.HTTPError(401, 'Please authorize')
```

Comment:
It would be nice to have JSON response for JSON request. Plz check whether it's possible to detect this.